xmodule/tests/test_item_bank.py

```diff
@@ -0,0 +1,498 @@
+                publish_item=True,
+            )
+            for i in range(4)
+        ]
+        self.publisher = Mock()  # for tests that look at analytics
+        self._reload_item_bank()
+
+    def _bind_course_block(self, block):
+        """
+        Bind a block (part of self.course) so we can access student-specific data.
+
+        (Not clear if this is necessary since XModules are all removed now. It's possible that this
+         could be removed without breaking tests.)
+        """
+        prepare_block_runtime(block.runtime, course_id=block.location.course_key)
+
+        def get_block(descriptor):
+            """Mocks module_system get_block function"""
+            prepare_block_runtime(descriptor.runtime, course_id=block.location.course_key)
+            descriptor.runtime.get_block_for_descriptor = get_block
+            descriptor.bind_for_student(self.user_id)
+            return descriptor
+
+        block.runtime.get_block_for_descriptor = get_block
+
+    def _reload_item_bank(self):
+        """
+        Reload self.item_bank. Do this if you want its `.children` list to be updated with what's in the db.
+
+        HACK: These test cases don't persist student state, but some tests need `selected` to persist betweem item_bank
+        reloads. So, we just transfer it from the old item_bank object instance to the new one, as if it were persisted.
+        """
+        selected = self.item_bank.selected
+        self.item_bank = self.store.get_item(self.item_bank.location)
+        self._bind_course_block(self.item_bank)
+        if selected:
+            self.item_bank.selected = selected
+        self.item_bank.runtime.publish = self.publisher
+
+
+@skip_unless_cms
+class TestItemBankForCms(ItemBankTestBase):
+    """
+    Test Studio ItemBank behaviors -- export/import, validation, author-facing views.
+    """
+    def test_xml_export_import_cycle(self):
+        """
+        Test the export-import cycle.
+        """
+        # Export self.item_bank to the virtual filesystem
+        export_fs = MemoryFS()
+        self.item_bank.runtime.export_fs = export_fs  # pylint: disable=protected-access
+        node = etree.Element("unknown_root")
+        self.item_bank.add_xml_to_node(node)
+
+        # Read back the itembank OLX
+        with export_fs.open('{dir}/{file_name}.xml'.format(
+            dir=self.item_bank.scope_ids.usage_id.block_type,
+            file_name=self.item_bank.scope_ids.usage_id.block_id
+        )) as f:
+            actual_olx_export = f.read()
+
+        # And compare.
+        expected_olx_export = (
+            '<itembank display_name="My Item Bank" max_count="1">\n'
+            '  <problem url_name="My_Item_0"/>\n'
+            '  <problem url_name="My_Item_1"/>\n'
+            '  <problem url_name="My_Item_2"/>\n'
+            '  <problem url_name="My_Item_3"/>\n'
+            '</itembank>\n'
+        )
+        self.maxDiff = None
```

Comment:
Nit: make this a static attribute of `ItemBankTestBase` perhaps?

Reply:
Done